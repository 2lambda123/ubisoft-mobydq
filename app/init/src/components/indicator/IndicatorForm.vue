<template>
  <div>
    <h1 class="mt-5">Edit Indicator</h1>

    <form>
      <div class="form-row">
        <div class="col-md-4">
          <!-- Data Source Form -->
          <div class="form-group required">
            <label for="indicatorName" class="col-form-label">
              Name:
            </label>
            <input
              class="form-control col-sm"
              id="indicatorName"
              type="text"
              required="required"
              placeholder="Type indicator name"
              v-model="indicator.name"
            />
          </div>
          <div class="form-group">
            <label for="indicatorDescription" class="col-form-label">
              Description:
            </label>
            <textarea
              class="form-control col-sm"
              id="indicatorDescription"
              placeholder="Type indicator description"
              rows="3"
              v-model="indicator.description"
            />
          </div>
          <div class="form-group required">
            <indicator-select-indicator-type
              v-model="indicator.indicatorTypeId"
              v-on:changeIndicatorType="getIndicatorType">
            </indicator-select-indicator-type>
          </div>
          <div class="form-group required">
            <indicator-select-indicator-group
              v-model="indicator.indicatorGroupId"
              v-on:changeIndicatorGroup="getIndicatorGroup">
            </indicator-select-indicator-group>
          </div>
          <div class="form-group">
            <label for="indicatorExecutionOrder" class="col-form-label">
              Execution Order:
            </label>
            <input
              class="form-control col-sm"
              type="number"
              id="indicatorExecutionOrder"
              v-model="indicator.executionOrder"
            />
          </div>
          <div class="custom-control custom-switch mr-4 mt-1 mb-2">
            <input class="custom-control-input" id="active" type="checkbox" value="" v-model="indicator.flagActive" />
            <label for="active" class="custom-control-label">
              Active
            </label>
          </div>

          <!-- Meta-Data -->
          <div>
            <indicator-meta-data
              v-if="indicator.id"
              v-bind:id="indicator.id"
              v-bind:createdDate="indicator.createdDate"
              v-bind:createdBy="indicator.userByCreatedById.email"
              v-bind:updatedDate="indicator.updatedDate"
              v-bind:updatedBy="indicator.userByUpdatedById.email"
            ></indicator-meta-data>
          </div>

          <!-- Button Menu -->
          <div class="mt-3">
            <indicator-button-save v-bind:indicator="indicator"> </indicator-button-save>
            <indicator-button-close> </indicator-button-close>
            <indicator-button-delete v-if="indicator.id" v-bind:indicatorId="indicator.id"> </indicator-button-delete>
          </div>
        </div>
      </div>
    </form>
  </div>
</template>

<script>
import IndicatorSelectIndicatorType from "./IndicatorSelectIndicatorType.vue";
import IndicatorSelectIndicatorGroup from "./IndicatorSelectIndicatorGroup.vue";
import IndicatorButtonSave from "./IndicatorButtonSave.vue";
import IndicatorButtonClose from "./IndicatorButtonClose.vue";
import IndicatorButtonDelete from "./IndicatorButtonDelete.vue";
import MetaDataCard from "../utils/MetaDataCard.vue";
import Mixins from "../utils/Mixins.vue";

export default {
  mixins: [Mixins],
  components: {
    "indicator-select-indicator-type": IndicatorSelectIndicatorType,
    "indicator-select-indicator-group": IndicatorSelectIndicatorGroup,
    "indicator-meta-data": MetaDataCard,
    "indicator-button-save": IndicatorButtonSave,
    "indicator-button-close": IndicatorButtonClose,
    "indicator-button-delete": IndicatorButtonDelete
  },
  data: function() {
    return {
      indicator: {}
    };
  },
  computed: {
    indicatorId() {
      return this.$route.params.indicatorId;
    }
  },
  created: function() {
    // If indicatorId != new then get data for existing indicator
    if (this.indicatorId != "new") {
      let payload = {
        query: this.$store.state.queryGetIndicator,
        variables: {
          id: parseInt(this.indicatorId)
        }
      };
      let headers = {};
      if (this.$session.exists()) {
        headers = { Authorization: "Bearer " + this.$session.get("jwt") };
      }
      this.$http.post(this.$store.state.graphqlUrl, payload, { headers }).then(
        function(response) {
          if (response.data.errors) {
            this.displayError(response);
          } else {
            this.indicator = response.data.data.indicatorById;
          }
        },
        // Error callback
        function(response) {
          this.displayError(response);
        }
      );
    }
  },
  methods: {
    getIndicatorType(value) {
      // Get indicator type from child component
      if (value != null) {
        this.indicator["indicatorTypeId"] = value;
      } else {
        this.indicator["indicatorTypeId"] = null;
      }
    },
    getIndicatorGroup(value) {
      // Get indicator group from child component
      if (value != null) {
        this.indicator["indicatorGroupId"] = value;
      } else {
        this.indicator["indicatorGroupId"] = null;
      }
    }
  }
};
</script>
