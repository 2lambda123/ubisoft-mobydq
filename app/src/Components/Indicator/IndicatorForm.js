import React from 'react';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import TextInput from './../FormInput/TextInput';
import SelectInput from './../FormInput/SelectInput';
import SwitchInput from './../FormInput/SwitchInput';
import SaveButton from './../FormInput/SaveButton';
import ExecuteButton from './../FormInput/ExecuteButton';

const IndicatorFormFields = (props) => {
  const {
    data,
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting
  } = props;
  return <form onSubmit={handleSubmit} style={{ 'marginLeft': '50px' }}>
    <div style={{ 'marginTop': '10px', 'marginBottom': '30px' }}>
      <SaveButton disabled={isSubmitting} />
      <ExecuteButton disabled={isSubmitting} />
    </div>
    <div>
      <TextInput
        id="name"
        label="Name"
        helperText=""
        placeholder=""
        touched={touched.name}
        error={touched.name && errors.name}
        value={values.name}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="description"
        label="Description"
        helperText=""
        placeholder=""
        touched={touched.description}
        error={touched.description && errors.description}
        value={values.description}
        onChange={handleChange}
        onBlur={handleBlur}
        multiline={true}
        rows={5}
      />
    </div>
    <div>
      <SelectInput
        id="indicatorTypeId"
        label="Type"
        items={data.allIndicatorTypes.nodes}
        touched={touched.indicatorTypeId}
        error={touched.indicatorTypeId && errors.indicatorTypeId}
        value={values.indicatorTypeId}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <SelectInput
        id="indicatorGroupId"
        label="Group"
        items={data.allIndicatorGroups.nodes}
        touched={touched.indicatorGroupId}
        error={touched.indicatorGroupId && errors.indicatorGroupId}
        value={values.indicatorGroupId}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="executionOrder"
        label="Execution Order"
        helperText=""
        numeric={true}
        touched={touched.executionOrder}
        error={touched.executionOrder && errors.executionOrder}
        value={values.executionOrder}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <SwitchInput
        id="flagActive"
        label="Active"
        touched={touched.flagActive}
        error={touched.flagActive && errors.flagActive}
        checked={values.flagActive}
        value="flagActive"
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="createdDate"
        label="Created Date"
        helperText=""
        placeholder=""
        touched={touched.createdDate}
        error={touched.createdDate && errors.createdDate}
        value={values.createdDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="createdBy"
        label="Created By"
        helperText=""
        placeholder=""
        touched={touched.createdBy}
        error={touched.createdBy && errors.createdBy}
        value={values.createdBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="updatedDate"
        label="Updated Date"
        helperText=""
        placeholder=""
        touched={touched.updatedDate}
        error={touched.updatedDate && errors.updatedDate}
        value={values.updatedDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="updatedBy"
        label="Updated By"
        helperText=""
        placeholder=""
        touched={touched.updatedBy}
        error={touched.updatedBy && errors.updatedBy}
        value={values.updatedBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
  </form>;
};

const formikEnhancer = withFormik({
  'validationSchema': Yup.object().shape({
    'indicatorTypeId': Yup.number().integer()
      .min(1, 'You need to select an indicator type')
      .required('You need to select an indicator type'),
    'indicatorGroupId': Yup.number().integer()
      .min(1, 'You need to select an indicator group')
      .required('You need to select an indicator group'),
    'executionOrder': Yup.number().integer()
      .min(0, 'Execution order has to be a non-negative integer')
      .required('You need to input execution order.'),
    'name': Yup.string()
      .required('Name cannot be blank'),
    'description': Yup.string()
      .required('Description cannot be blank')
  }),

  'mapPropsToValues': (props) => props.initialFieldValues === null
    ? {
        'name': '',
        'description': '',
        'executionOrder': 0,
        'indicatorTypeId': 0,
        'indicatorGroupId': 0,
        'flagActive': false
      }
    : {
        'name': props.initialFieldValues.name,
        'description': props.initialFieldValues.description,
        'executionOrder': props.initialFieldValues.executionOrder,
        'indicatorTypeId': props.initialFieldValues.indicatorTypeId,
        'indicatorGroupId': props.initialFieldValues.indicatorGroupId,
        'flagActive': props.initialFieldValues.flagActive,
        'createdDate': props.initialFieldValues.createdDate,
        'createdBy': props.initialFieldValues.userByCreatedById.email,
        'updatedDate': props.initialFieldValues.updatedDate,
        'updatedBy': props.initialFieldValues.userByUpdatedById.email
      },
  'handleSubmit': (payload, { props, setSubmitting }) => {
    setSubmitting(false);
    let variables;
    if (props.initialFieldValues === null) {
      variables = { 'indicator': payload };
    } else {
      variables = { 'indicatorPatch': payload, 'id': props.initialFieldValues.id };
    }
    props.mutate({
      variables
    });
  },
  'displayName': 'IndicatorForm'
});


const EnhancedIndicatorForm = formikEnhancer(IndicatorFormFields);

export default EnhancedIndicatorForm;
