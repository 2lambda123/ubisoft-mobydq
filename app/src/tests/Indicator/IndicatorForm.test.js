import React from 'react';
import { shallowWrap } from './../../setupTests';

import IndicatorForm from './../../Components/Indicator/IndicatorForm';

describe('IndicatorForm component unit test', () => {
  const data = {
    'allIndicatorGroups': {
      'nodes': [
        {
          'id': 1,
          'value': 'mock indicator group'
        }
      ]
    },
    'allIndicatorTypes': {
      'nodes': [
        {
          'id': 1,
          'value': 'mock indicator type'
        }
      ]
    }
  };

  const wrapper = shallowWrap(<IndicatorForm data={data} initialFieldValues={null} handleBlur={jest.fn()} handleChange={jest.fn()} handleSubmit={jest.fn()} isSubmitting={false}/>);
  it('matches snapshot', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
