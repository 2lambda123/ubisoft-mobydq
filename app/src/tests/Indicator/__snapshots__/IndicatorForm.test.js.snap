// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndicatorForm component unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Formik
    data={
      Object {
        "allIndicatorGroups": Object {
          "nodes": Array [
            Object {
              "id": 1,
              "value": "mock indicator group",
            },
          ],
        },
        "allIndicatorTypes": Object {
          "nodes": Array [
            Object {
              "id": 1,
              "value": "mock indicator type",
            },
          ],
        },
      }
    }
    displayName="IndicatorForm"
    enableReinitialize={false}
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleSubmit={[Function]}
    initialFieldValues={null}
    initialValues={
      Object {
        "description": "",
        "executionOrder": 0,
        "flagActive": false,
        "indicatorGroupId": 0,
        "indicatorTypeId": 0,
        "name": "",
      }
    }
    isInitialValid={false}
    isSubmitting={false}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <IndicatorFormFields
        data={
          Object {
            "allIndicatorGroups": Object {
              "nodes": Array [
                Object {
                  "id": 1,
                  "value": "mock indicator group",
                },
              ],
            },
            "allIndicatorTypes": Object {
              "nodes": Array [
                Object {
                  "id": 1,
                  "value": "mock indicator type",
                },
              ],
            },
          }
        }
        dirty={false}
        errors={Object {}}
        handleBlur={[Function]}
        handleChange={[Function]}
        handleReset={[Function]}
        handleSubmit={[Function]}
        initialFieldValues={null}
        initialValues={
          Object {
            "description": "",
            "executionOrder": 0,
            "flagActive": false,
            "indicatorGroupId": 0,
            "indicatorTypeId": 0,
            "name": "",
          }
        }
        isSubmitting={false}
        isValid={false}
        isValidating={false}
        registerField={[Function]}
        resetForm={[Function]}
        setError={[Function]}
        setErrors={[Function]}
        setFieldError={[Function]}
        setFieldTouched={[Function]}
        setFieldValue={[Function]}
        setFormikState={[Function]}
        setStatus={[Function]}
        setSubmitting={[Function]}
        setTouched={[Function]}
        setValues={[Function]}
        submitCount={0}
        submitForm={[Function]}
        touched={Object {}}
        unregisterField={[Function]}
        validateField={[Function]}
        validateForm={[Function]}
        validateOnBlur={true}
        validateOnChange={true}
        values={
          Object {
            "description": "",
            "executionOrder": 0,
            "flagActive": false,
            "indicatorGroupId": 0,
            "indicatorTypeId": 0,
            "name": "",
          }
        }
      />,
      "value": Object {
        "dirty": false,
        "errors": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleReset": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "description": "",
          "executionOrder": 0,
          "flagActive": false,
          "indicatorGroupId": 0,
          "indicatorTypeId": 0,
          "name": "",
        },
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "registerField": [Function],
        "resetForm": [Function],
        "setError": [Function],
        "setErrors": [Function],
        "setFieldError": [Function],
        "setFieldTouched": [Function],
        "setFieldValue": [Function],
        "setFormikState": [Function],
        "setStatus": [Function],
        "setSubmitting": [Function],
        "setTouched": [Function],
        "setValues": [Function],
        "status": undefined,
        "submitCount": 0,
        "submitForm": [Function],
        "touched": Object {},
        "unregisterField": [Function],
        "validate": undefined,
        "validateField": [Function],
        "validateForm": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": [Function],
        "values": Object {
          "description": "",
          "executionOrder": 0,
          "flagActive": false,
          "indicatorGroupId": 0,
          "indicatorTypeId": 0,
          "name": "",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {
          "allIndicatorGroups": Object {
            "nodes": Array [
              Object {
                "id": 1,
                "value": "mock indicator group",
              },
            ],
          },
          "allIndicatorTypes": Object {
            "nodes": Array [
              Object {
                "id": 1,
                "value": "mock indicator type",
              },
            ],
          },
        },
        "dirty": false,
        "errors": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleReset": [Function],
        "handleSubmit": [Function],
        "initialFieldValues": null,
        "initialValues": Object {
          "description": "",
          "executionOrder": 0,
          "flagActive": false,
          "indicatorGroupId": 0,
          "indicatorTypeId": 0,
          "name": "",
        },
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "registerField": [Function],
        "resetForm": [Function],
        "setError": [Function],
        "setErrors": [Function],
        "setFieldError": [Function],
        "setFieldTouched": [Function],
        "setFieldValue": [Function],
        "setFormikState": [Function],
        "setStatus": [Function],
        "setSubmitting": [Function],
        "setTouched": [Function],
        "setValues": [Function],
        "status": undefined,
        "submitCount": 0,
        "submitForm": [Function],
        "touched": Object {},
        "unregisterField": [Function],
        "validateField": [Function],
        "validateForm": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "values": Object {
          "description": "",
          "executionOrder": 0,
          "flagActive": false,
          "indicatorGroupId": 0,
          "indicatorTypeId": 0,
          "name": "",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {},
        "_currentValue2": Object {},
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <IndicatorFormFields
          data={
            Object {
              "allIndicatorGroups": Object {
                "nodes": Array [
                  Object {
                    "id": 1,
                    "value": "mock indicator group",
                  },
                ],
              },
              "allIndicatorTypes": Object {
                "nodes": Array [
                  Object {
                    "id": 1,
                    "value": "mock indicator type",
                  },
                ],
              },
            }
          }
          dirty={false}
          errors={Object {}}
          handleBlur={[Function]}
          handleChange={[Function]}
          handleReset={[Function]}
          handleSubmit={[Function]}
          initialFieldValues={null}
          initialValues={
            Object {
              "description": "",
              "executionOrder": 0,
              "flagActive": false,
              "indicatorGroupId": 0,
              "indicatorTypeId": 0,
              "name": "",
            }
          }
          isSubmitting={false}
          isValid={false}
          isValidating={false}
          registerField={[Function]}
          resetForm={[Function]}
          setError={[Function]}
          setErrors={[Function]}
          setFieldError={[Function]}
          setFieldTouched={[Function]}
          setFieldValue={[Function]}
          setFormikState={[Function]}
          setStatus={[Function]}
          setSubmitting={[Function]}
          setTouched={[Function]}
          setValues={[Function]}
          submitCount={0}
          submitForm={[Function]}
          touched={Object {}}
          unregisterField={[Function]}
          validateField={[Function]}
          validateForm={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
          values={
            Object {
              "description": "",
              "executionOrder": 0,
              "flagActive": false,
              "indicatorGroupId": 0,
              "indicatorTypeId": 0,
              "name": "",
            }
          }
        />,
        "value": Object {
          "dirty": false,
          "errors": Object {},
          "handleBlur": [Function],
          "handleChange": [Function],
          "handleReset": [Function],
          "handleSubmit": [Function],
          "initialValues": Object {
            "description": "",
            "executionOrder": 0,
            "flagActive": false,
            "indicatorGroupId": 0,
            "indicatorTypeId": 0,
            "name": "",
          },
          "isSubmitting": false,
          "isValid": false,
          "isValidating": false,
          "registerField": [Function],
          "resetForm": [Function],
          "setError": [Function],
          "setErrors": [Function],
          "setFieldError": [Function],
          "setFieldTouched": [Function],
          "setFieldValue": [Function],
          "setFormikState": [Function],
          "setStatus": [Function],
          "setSubmitting": [Function],
          "setTouched": [Function],
          "setValues": [Function],
          "status": undefined,
          "submitCount": 0,
          "submitForm": [Function],
          "touched": Object {},
          "unregisterField": [Function],
          "validate": undefined,
          "validateField": [Function],
          "validateForm": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
          "validationSchema": [Function],
          "values": Object {
            "description": "",
            "executionOrder": 0,
            "flagActive": false,
            "indicatorGroupId": 0,
            "indicatorTypeId": 0,
            "name": "",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "allIndicatorGroups": Object {
              "nodes": Array [
                Object {
                  "id": 1,
                  "value": "mock indicator group",
                },
              ],
            },
            "allIndicatorTypes": Object {
              "nodes": Array [
                Object {
                  "id": 1,
                  "value": "mock indicator type",
                },
              ],
            },
          },
          "dirty": false,
          "errors": Object {},
          "handleBlur": [Function],
          "handleChange": [Function],
          "handleReset": [Function],
          "handleSubmit": [Function],
          "initialFieldValues": null,
          "initialValues": Object {
            "description": "",
            "executionOrder": 0,
            "flagActive": false,
            "indicatorGroupId": 0,
            "indicatorTypeId": 0,
            "name": "",
          },
          "isSubmitting": false,
          "isValid": false,
          "isValidating": false,
          "registerField": [Function],
          "resetForm": [Function],
          "setError": [Function],
          "setErrors": [Function],
          "setFieldError": [Function],
          "setFieldTouched": [Function],
          "setFieldValue": [Function],
          "setFormikState": [Function],
          "setStatus": [Function],
          "setSubmitting": [Function],
          "setTouched": [Function],
          "setValues": [Function],
          "status": undefined,
          "submitCount": 0,
          "submitForm": [Function],
          "touched": Object {},
          "unregisterField": [Function],
          "validateField": [Function],
          "validateForm": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
          "values": Object {
            "description": "",
            "executionOrder": 0,
            "flagActive": false,
            "indicatorGroupId": 0,
            "indicatorTypeId": 0,
            "name": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {},
          "_currentValue2": Object {},
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
